### PR TITLE
Update sdk for conversation and discover to 3.8 to fix bluemix problem

### PR DESCRIPTION
**Pull request by stevenpkg**
@germanattanasio this is an urgent fix for bluemix sdk update

### CODE REVIEW

File: src/main/java/com/ibm/watson/apis/conversation_with_discovery/listener/SetupThread.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ public void run() {
 
       status = "Conversation ";
 
-      ConversationService service = new ConversationService(ConversationService.VERSION_DATE_2016_07_11);
+      ConversationService service = new ConversationService(Constants.CONVERSATION_VERSION);
```
Comment (germanattanasio) [inline src/main/java/com/ibm/watson/apis/conversation_with_discovery/listener/SetupThread.java:142]: Use the constant from the SDK.

File: src/test/java/com/ibm/watson/apis/conversation_with_discovery/rest/ProxyResourceTest.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ public void tearDown() throws IOException {
    * @throws IOException Signals that an I/O exception has occurred.
    * @throws InterruptedException the 4interrupted exception
    */
-  @Test
+  //@Test
   public void testSendMessage() throws IOException, InterruptedException {
```
Comment (germanattanasio) [inline src/test/java/com/ibm/watson/apis/conversation_with_discovery/rest/ProxyResourceTest.java]: Why are you commenting this line?
Can you add tests cases for the changes you are doing?

File: build.gradle [hunk 3]
```diff
@@ -49,8 +49,8 @@ dependencies {
     compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.1'
     compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.3.2'
     compile group: 'org.jsoup', name: 'jsoup', version: '1.9.2'
-    compile group: 'com.ibm.watson.developer_cloud', name: 'conversation', version:'3.5.3'
-    compile group: 'com.ibm.watson.developer_cloud', name: 'discovery', version:'3.5.3'
+    compile group: 'com.ibm.watson.developer_cloud', name: 'conversation', version:'3.8.0'
+    compile group: 'com.ibm.watson.developer_cloud', name: 'discovery', version:'3.8.0'
```
Comment (germanattanasio) [inline build.gradle:53]: `okhttp` should be updated to 3.8.0

**Review by germanattanasio — CHANGES_REQUESTED** [review]
Hi @stevenpkg, I left some comments, can you also add some test cases?

File: build.gradle (hunk 3 above)
Comment (stevenpkg) [reply build.gradle:53]: //@Test
public void testSendMessage() throws IOException, InterruptedException {
       Test is broken with 3.8.0 update, and I needed to get a fix out quickly, will fix very soon

Use the constant from the SDK
      It does, just keeps it in one place Constants.java

Comment (stevenpkg) [reply build.gradle:53]: Fixed testSendMessage() by updating mockwebserver to 3.8.0

**Comment by stevenpkg** [comment]
@germanattanasio can you please approve

**Review by germanattanasio — APPROVED** [review]

**Comment by germanattanasio** [comment]
@stevenpkg you should make sure that the dependencies listed below are updated

```
    compile group: 'org.apache.solr', name: 'solr-solrj', version: '5.4.1'
    compile group: 'com.squareup.okhttp3', name: 'okhttp', version: '3.3.1'
    compile group: 'com.squareup.okhttp3', name: 'logging-interceptor', version: '3.3.1'
    compile group: 'com.squareup.okhttp3', name: 'okhttp-ws', version: '3.3.1'
    compile group: 'com.squareup.okhttp3', name: 'okhttp-urlconnection', version: '3.3.1'
    compile group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.3.6'
    compile group: 'org.apache.httpcomponents', name: 'httpcore', version: '4.3.2'
```

They all have problems with the new IBM JDK. We are doing a change to the SSL certificates we use and it will break this app after July 24 if you don't update the dependencies

**Comment by stevenpkg** [comment]
@germanattanasio will update dependencies you listed

**Pull request merged**